src/StreamSocket.js

```diff
@@ -0,0 +1,62 @@
+const Transform = require('stream').Transform;
+var d1 = require('diya-sdk');
+
+class StreamSocket extends Transform {
+	constructor(d1inst, params, options) {
+		super(options);
+		this.d1inst = d1inst;
+		this.socketName = params.socketName;
+		this.socketId = params.socketId;
+		this.peerId = params.target;
+		this.flagSocketIsDead = false;
+		this.subscriptionSocketClosed = null;
+	}
+
+	_read() {
+	}
+
+	_write(chunk, encoding, callback) {
+		if (this.flagSocketIsDead === false) {
+			var params = {
+				data: {
+					socketName: this.socketName,
```

Comment:
I think only ```socketId``` is useful, no need to send ```socketName``` every single time...
Keep in mine that this communication channel should stay as low overhead as possible just send what's absolutely necessary